### PR TITLE
V6 tee update

### PR DESCRIPTION
**Pull request by PierreJeanjacquot**
Build, Deploy & Run an iExec v6 TEE debug app (with tips & tricks)

preview: https://documentations.iex.ec/staging/v/v6-tee-update/for-developers/confidential-computing/create-your-first-sgx-app

### CODE REVIEW

**Comment by PierreJeanjacquot** [comment]
reviews welcome, merge target 19/07/21!

File: for-developers/confidential-computing/create-your-first-sgx-app.md [hunk 1]
```diff
@@ -24,188 +29,186 @@ cd ~/iexec-projects
...
+
+# pull the SCONE currated image corresponding to our base image
+docker pull registry.scontain.com:5050/sconecuratedimages/node:14.4.0-alpine3.11
+
+# run the sconifier to build the TEE image based on the non-TEE image
+docker run -it --rm \
+            -v /var/run/docker.sock:/var/run/docker.sock \
+            registry.scontain.com:5050/scone-production/iexec-sconify-image:5.3.7 \
+            sconify_iexec \
+            --name=hello-world \
```
Comment (luisccastillo) [inline for-developers/confidential-computing/create-your-first-sgx-app.md]: Fixed value, should it be ${IMG_NAME} ?

Comment (PierreJeanjacquot) [reply for-developers/confidential-computing/create-your-first-sgx-app.md]: good question `--name` is not documented, @jeremyjams thought?

Comment (PierreJeanjacquot) [reply for-developers/confidential-computing/create-your-first-sgx-app.md]: @luisccastillo I'll change it :+1:

File: for-developers/confidential-computing/create-your-first-sgx-app.md [hunk 2]
```diff
@@ -24,188 +29,186 @@ cd ~/iexec-projects
...
+  ...
+  "app": {
+    "owner": "0xF048eF3d7E3B33A465E0599E641BB29421f7Df92", // your address
+    "name": "tee-hello-world", // application name
+    "type": "DOCKER",
+    "multiaddr": "docker.io/username/my-tee-hello-world:1.0.0", // app image
+    "checksum": "0x15bed530c76f1f3b05b2db8d44c417128b8934899bc85804a655a01b441bfa78", // image digest
+    "mrenclave": {
+      "provider": "SCONE", // TEE provider (keep default value)
+      "version": "v5", // protocol version (keep default value)
```
Comment (james-toussaint) [inline for-developers/confidential-computing/create-your-first-sgx-app.md]:
```suggestion
      "version": "v5", // Scone version (keep default value)
```

File: for-developers/confidential-computing/create-your-first-sgx-app.md [hunk 3]
```diff
@@ -24,188 +29,186 @@ cd ~/iexec-projects
...
+      "entrypoint": "node /app/app.js", // your app image entrypoint
+      "heapSize": 1073741824, // heap size in bytes (1Gb)
```
Comment (james-toussaint) [inline for-developers/confidential-computing/create-your-first-sgx-app.md]:
```suggestion
      "heapSize": 1073741824, // heap size in bytes (1GB)
```

File: for-developers/confidential-computing/create-your-first-sgx-app.md [hunk 4]
```diff
@@ -24,188 +29,186 @@ cd ~/iexec-projects
...
+      "fingerprint": "eca3ace86f1e8a5c47123c8fd271319e9eb25356803d36666dc620f30365c0c1" // fspf fingerprint see below
```
Comment (james-toussaint) [inline for-developers/confidential-computing/create-your-first-sgx-app.md]:
```suggestion
      "fingerprint": "eca3ace86f1e8a5c47123c8fd271319e9eb25356803d36666dc620f30365c0c1" // fingerprint of the enclave code (mrenclave), see how to retrieve it below
```

File: for-developers/confidential-computing/create-your-first-sgx-app.md [hunk 5]
```diff
@@ -24,188 +29,186 @@ cd ~/iexec-projects
...
   ...
- "mrenclave": "8a2e59370e47425ebaad0ba72ab06beb49ddf53aa1575c0de9a32dc82687d20c|695e1fd6bb78cc6745786d9941dda921|a8e434c81b82012c19d028ab3e7ef3adecb7786c10e5739422a7f7444e2d323c|node /app/app.js"
 }
-#####################################################################
-Hint: Replace 'mrenclave' before doing 'iexec app deploy' step.
 ```
 
-That alphanumeric string is the [fingerprint](intel-sgx-technology.md#applications-fingerprint) of your application. It allows the verification of its integrity.
+{% hint style="info" %}
+Run your TEE image with `SCONE_HASH=1` to get the fspf fingerprint:
```
Comment (james-toussaint) [inline for-developers/confidential-computing/create-your-first-sgx-app.md]:
```suggestion
Run your TEE image with `SCONE_HASH=1` to get the enclave fingerprint (mrenclave):
```

File: for-developers/confidential-computing/create-your-first-sgx-app.md [hunk 6]
```diff
@@ -24,190 +29,187 @@ cd ~/iexec-projects
...
 mkdir src
 touch Dockerfile
-touch protect-fs.sh
+touch sconify.sh
 ```
 
-In the folder `src/` create the file `app.js` \(or `app.py` if you want to use Python\) then copy [this](../your-first-app.md#write-the-app-shell-script-example) code inside.
+In the folder `src/` create the file `app.js` then copy [this](../your-first-app#write-the-app-javascript-script-example) code inside.
+
+As we mentioned earlier, the advantage of using **SCONE** is the ability to make the application **Intel® SGX-enabled** without changing the source code. The only thing we are going to do is rebuilding the app using the Trusted-Execution-Environment currated image provided by **SCONE**.
```
Comment (james-toussaint) [inline for-developers/confidential-computing/create-your-first-sgx-app.md]:
```suggestion
As we mentioned earlier, the advantage of using **SCONE** is the ability to make the application **Intel® SGX-enabled** without changing the source code. The only thing we are going to do is rebuilding the app using the Trusted-Execution-Environment tooling provided by **SCONE**.
```

File: for-developers/confidential-computing/create-your-first-sgx-app.md [hunk 7]
```diff
@@ -24,190 +29,187 @@ cd ~/iexec-projects
...
+
+{% hint style="info" %}
+SCONE provides currated base images for a bunch of languages, the currated image will be used to replace the base image and get rid of any non TEE compliant operations.
```
Comment (james-toussaint) [inline for-developers/confidential-computing/create-your-first-sgx-app.md]:
```suggestion
SCONE provides TEE conversion tooling (Python, Java, ..) plus eventually TEE base images for other languages (NodeJs).
```

File: for-developers/confidential-computing/create-your-first-sgx-app.md [hunk 8]
```diff
@@ -24,190 +29,187 @@ cd ~/iexec-projects
...
 
-{% tab title="Python" %}
-{% code title="Dockefile" %}
-```bash
-FROM sconecuratedimages/public-apps:python-3.7.3-alpine3.10-scone3.0
+## Build the TEE docker image:
 
-### install python3 dependencies you need
-RUN SCONE_MODE=sim pip3 install pyfiglet
+You will need to register a [free SCONE Account](https://scontain.com) to access the SCONE build tools and currated images from the [SCONE registry](https://gitlab.scontain.com/).
```
Comment (james-toussaint) [inline for-developers/confidential-computing/create-your-first-sgx-app.md]:
```suggestion
You will need to register a [free SCONE Account](https://scontain.com) to access the SCONE build tools and curated images from the [SCONE registry](https://gitlab.scontain.com/).
```

File: for-developers/confidential-computing/create-your-first-sgx-app.md [hunk 9]
```diff
@@ -24,190 +29,187 @@ cd ~/iexec-projects
...
+            --fs-dir=/app \
+            --host-path=/etc/hosts \
+            --host-path=/etc/resolv.conf \
+            --binary=/usr/local/bin/node \
+            --heap=1G \
+            --dlopen=2 \
+            --no-color \
+            --verbose \
+            --command=${ENTRYPOINT} \
+            && echo && echo "successfully built TEE docker image => ${IMG_TO}"
```
Comment (james-toussaint) [inline for-developers/confidential-computing/create-your-first-sgx-app.md]: Maybe print SCONE_HASH here?

File: for-developers/confidential-computing/create-your-first-sgx-app.md [hunk 10]
```diff
@@ -24,190 +29,187 @@ cd ~/iexec-projects
...
+
+Specify the tag `--tag tee` in `iexec app run` command to run a tee app.
+
+
+One last thing, in order to run a **TEE-debug** app you will also need to select a debug workerpool, use the Viviani debug workerpool `0xe6806E69BA8650AF23264702ddD43C4DCe35CcCe` (see deployed workerpools on https://v6.pools.iex.ec).
 
-* When you tag your Docker image, use `my-tee-hello-world` instead of `my-hello-world`.
-* Do not forget the put the application's fingerprint inside `iexec.json` \(`"mrenclave"` attribute\).
-* Add the option `--tag tee` to the command `iexec app run`.
+The debug workerpool is connected to a debug Secret Management System (this is fine for debuging but do not use to store production secrets), we will need to init the storage token on this SMS.
```
Comment (james-toussaint) [inline for-developers/confidential-computing/create-your-first-sgx-app.md]:
```suggestion
The debug workerpool is connected to a debug Secret Management Service (this is fine for debugging but do not use to store production secrets), we will need to init the storage token on this SMS.
```

File: for-developers/confidential-computing/create-your-first-sgx-app.md (hunk 9 above)
Comment (PierreJeanjacquot) [reply for-developers/confidential-computing/create-your-first-sgx-app.md]: yes, added: f44792fe5755e07134a6034746833b845971fc36

**Comment by PierreJeanjacquot** [comment]
need 1 approval to merge

File: for-developers/confidential-computing/create-your-first-sgx-app.md [hunk 11]
```diff
@@ -37,7 +37,7 @@ In the folder `src/` create the file `app.js` then copy [this](../your-first-app
 As we mentioned earlier, the advantage of using **SCONE** is the ability to make the application **Intel® SGX-enabled** without changing the source code. The only thing we are going to do is rebuilding the app using the Trusted-Execution-Environment tooling provided by **SCONE**.
 
 {% hint style="info" %}
-SCONE provides currated base images for a bunch of languages, the currated image will be used to replace the base image and get rid of any non TEE compliant operations.
+SCONE provides TEE conversion tooling (Python, Java, ..) plus eventually TEE base images for other languages (NodeJs).
```
Comment (ericr6) [inline for-developers/confidential-computing/create-your-first-sgx-app.md:41]: not sure to understand the second part

**Review by ericr6 — APPROVED** [review]

**Pull request merged**